### PR TITLE
do not create & store context in runtime vm

### LINKED ISSUE #4634: runtime VM holds context
which is not idiomatic: https://golang.org/pkg/context/ (it's possible we do it in other places)

we should remove the context internally, and rework the runtime API to be passed a context when it's needed

### PR DESCRIPTION
**Pull request by pperiyasamy**
Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

#### What type of PR is this?

> /kind bug

#### What this PR does / why we need it:

This PR reuses existing context object and avoids creating a new context object in runtime VM module and storing in its runtimeVM structure.

#### Which issue(s) this PR fixes:

The runtime VM creates its own context object and store in its structure which shouldn't be done as for crio process should contain only one context object which should be passed across API boundaries.  

Fixes #4634

#### Special notes for your reviewer:

`streaming#criAdapter{} `needed to store the existing `ctx` object, is that ok ?

#### Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

File: internal/oci/runtime_vm.go
Comment (haircommander) [inline internal/oci/runtime_vm.go:155]: why this change? also, why aren't we passing the ctx to CreateContainer as well?

**Comment by haircommander** [comment]
awesome thanks for taking this on @pperiyasamy , you've got a lint issue and I've got one question, but generally it LGTM

File: internal/oci/runtime_vm.go
Comment (pperiyasamy) [reply internal/oci/runtime_vm.go]: yes, missed to replace it with ctx, fixed now. 
ctx is updated with Namespace in startRuntimeDaemon, so its not needed anymore in other places. is my understanding right ?

File: server/streaming/streaming.go
Comment (pperiyasamy) [inline server/streaming/streaming.go:66]: while fixing lint in [suite_test.go](https://github.com/cri-o/cri-o/blob/master/server/suite_test.go#L173), this change has a compatibility issue with kubelet interface https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cri/streaming/server.go#L63 as it doesn't pass ctx object, how would like to fix this ?

File: server/streaming/streaming.go
Comment (pperiyasamy) [inline server/streaming/streaming.go:367]: @haircommander is it ok to store ctx here ?

File: server/streaming/streaming.go
Comment (haircommander) [reply server/streaming/streaming.go:66]: damn yeah we'll have to drop them here and store them for this...

File: server/streaming/streaming.go
Comment (haircommander) [reply server/streaming/streaming.go:367]: yeah I think we're going to have to. do we create a criAdapter for each new request or for each runtime?

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests.

**Comment by haircommander** [comment]
/approve

LGTM, thanks @pperiyasamy !

**Comment by haircommander** [comment]
@mrunalp @saschagrunert @umohnani8 @kolyshkin @giuseppe @QiWang19 PTAL

**Comment by fidencio** [comment]
@pperiyasamy, thanks a lot for the PR. Codewise it looks good to me. 

One thing I'd like to ask you for is, could you squash those 3 commits into a single one?  The reason I'm asking for this is, when we end up bisecting something, we don't end up in a commit breaking the tests, leaving us without knowing whether the commit is the reason of the breakage we're debugging, or just something that had a fix pushed apart.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by fidencio** [comment]
@saschagrunert, sorry, I'm holding this one till we have all patches squashed.  Once we have that, I'll proceed with the `/lgtm`.

**Comment by fidencio** [comment]
@saschagrunert taught me a nice trick: `/label tide/merge-method-squash`, let's see if it works. :-)

**Comment by fidencio** [comment]
oh, too late it seems. :-)

**Pull request merged**
